plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -333,10 +335,20 @@ protected static Optional<ColumnMapping> mapToUnboundedVarchar(JdbcTypeHandle ty
         return Optional.of(ColumnMapping.sliceMapping(
                 unboundedVarcharType,
                 varcharReadFunction(unboundedVarcharType),
-                (statement, index, value) -> {
-                    throw new TrinoException(
-                            NOT_SUPPORTED,
-                            "Underlying type that is mapped to VARCHAR is not supported for INSERT: " + typeHandle.getJdbcTypeName().get());
+                new SliceWriteFunction() {
+                    @Override
+                    public void setNull(PreparedStatement statement, int index) throws SQLException
```

Comment:
nit: (here and throughout) - should `setNull` be after `set` ?
The main purpose of the Write function is the `set` - the `setNull` is an optimization (and should follow it alphabetically in any case).